packages/platform-server/test/integration_spec.ts

```diff
@@ -1340,6 +1348,43 @@ class HiddenModule {}
         });
       });
 
+      describe('detecting state being transferred twice', () => {
+        it(`shows a warning when server providers has been provided twice`, async () => {
+          const consoleSpy = spyOn(console, 'warn');
+          const options = {document: '<app></app>'};
+          const bootstrap = renderModule(DoubleTransferStateModule, options);
+
+          // Note: script#ng-state repeated twice below
+          const expectedOutput =
+            '<html><head></head><body><app ng-version="0.0.0-PLACEHOLDER" ng-server-context="other"><div>Works!</div></app>' +
+            '<script id="ng-state" type="application/json">{"some-key":"some-value"}</script><script id="ng-state" type="application/json">{"some-key":"some-value"}</script></body></html>';
+          const output = await bootstrap;
+          expect(output).toEqual(expectedOutput);
+          expect(consoleSpy).toHaveBeenCalledWith(
+            jasmine.stringMatching('Angular detected an incompatible configuration'),
+          );
+          expect(consoleSpy).toHaveBeenCalledWith(
+            jasmine.stringMatching(
+              `This can happen if the server providers have been provided more than once using different mechanisms.`,
+            ),
+          );
+        });
+
+        it(`shows a warning when server providers has been provided twice`, async () => {
```

Comment:
Would this test description be more accurate?
```suggestion
        it(`should not show a warning when server providers were provided once`, async () => {
```